File: VisualDX/WinDXui/DisconnectFromServerCommand.cs
// Complete 4/24/2006

using System;
using System.Collections.Generic;
using System.Text;

namespace WinDX.UI
{
    class DisconnectFromServerCommand : ConfirmedCommand
    {
        public DisconnectFromServerCommand(String name,
            CommandScope scope, bool active)
            :
            base(name, scope, active,
            "Disconnect from Server",
            "Do you really want to disconnect from the server?") { }

        public override bool doIt(CommandInterface ci)
        {
            return DXApplication.theDXApplication.disconnectFromServer();
        }

    }
}

File: dx/src/uipp/windows/DisconnectFromServerCommand.cpp
/***********************************************************************/
/* Open Visualization Data Explorer                                    */
/* (C) Copyright IBM Corp. 1989,1999                                   */
/* ALL RIGHTS RESERVED                                                 */
/* This code licensed under the                                        */
/*    "IBM PUBLIC LICENSE - Open Visualization Data Explorer"          */
/***********************************************************************/

#include <dxconfig.h>
#include "defines.h"



#include "DisconnectFromServerCommand.h"
#include "DXApplication.h"


DisconnectFromServerCommand::DisconnectFromServerCommand(const char*   name,
                CommandScope* scope,
                bool       active):
     ConfirmedCommand(name, scope, active,
		      "Disconnect from Server...",
		      "Do you really want to disconnect from the server?")
{
}

DisconnectFromServerCommand::~DisconnectFromServerCommand()
{
}

bool DisconnectFromServerCommand::doIt(CommandInterface *ci)
{
    return theDXApplication->disconnectFromServer();
}
